packages/medusa/src/api/routes/store/carts/create-line-item/utils/handler-steps.ts

```diff
@@ -87,3 +88,88 @@ export async function handleAddOrUpdateLineItem(
     response_body: { cart },
   }
 }
+
+export async function addOrUpdateLineItem({
+  cartId,
+  container,
+  manager,
+  data,
+}) {
+  const cartService: CartService = container.resolve("cartService")
+  const lineItemService: LineItemService = container.resolve("lineItemService")
+
+  const cart = await cartService.retrieve(cartId, {
+    select: ["id", "region_id", "customer_id"],
+  })
+
+  const line = await lineItemService
+    .withTransaction(manager)
+    .generate(data.variant_id, cart.region_id, data.quantity, {
+      customer_id: data.customer_id || cart.customer_id,
+      metadata: data.metadata,
+    })
+
+  await manager.transaction(async (transactionManager) => {
+    const txCartService = cartService.withTransaction(transactionManager)
+
+    await txCartService.addOrUpdateLineItems(cart.id, line, {
+      validateSalesChannels:
+        featureFlagRouter.isFeatureEnabled("sales_channels"),
+    })
+  })
+}
+
+export async function setPaymentSessionAndVariantAvailability({
```

Comment:
thought: Should we separate this into two functions? I imagine this will live as two steps in the workflow in the future, so it would make for an easier migration.

Reply:
I realise we are "collecting" promises here, which is good for performance, so I am also fine with keeping it as is

Reply:
done